src/controllers/loginController.js

```diff
@@ -1,31 +1,20 @@
-const loginService = require('../services/registration');
+class LoginController {
+  constructor(loginService) {
+    this.loginService = loginService;
+    this.login = this.login.bind(this);
+  }
 
-const loginController = (req, res) => {
-  if (req.body.username && req.body.password) {
-    const reqUsernamePw = {
-      username: req.body.username,
-      password: req.body.password,
-    };
-    loginService.authorizeUser(reqUsernamePw)
-      .then((data) => {
-        const dataJSON = {
-          userId: data,
-        };
-        res.status(200).json(dataJSON);
-      }).catch((error) => {
-        error = {
-          message: 'Incorrect Username and/or Password!',
-        };
-        res.status(400).send(error);
-      });
-  } else {
-    const notEnteredDataError = {
-      message: 'Please enter a Username and a Password!',
-    };
-    res.status(400).send(notEnteredDataError);
+  login(req, res) {
+    if (req.body.username && req.body.password) {
+      this.loginService.authorizeUser({
+        username: req.body.username,
+        password: req.body.password,
+      }).then(data => res.status(200).json({ userId: data }))
```

Comment:
the answer object should not be created in the controller unless you want to attach metadata to it. if you do this then you wont be able to change the data structure so easily.

Reply:
corrected

Reply:
wrong line, wrong correction, the service call was fine, i commented the 12th line